pkg/sideinput/interface.go

```diff
@@ -0,0 +1,19 @@
+package sideinput
+
+import (
+	"context"
+)
+
+// SideInputRetriever is the interface for side input retrieval implementation.
+type SideInputRetriever interface {
+	// RetrieveSideInput is the function to process each side-input request.
+	RetrieveSideInput(ctx context.Context) (Message, error)
```

Comment:
return `(*Message, error)`?

Reply:
let's not return error. what is numaflow going to do if we see an error?

Reply:
We would just log it right now!
```
if err != nil {
return fmt.Errorf("failed to retrieve side input: %w", err)
}
```

Reply:
Also @whynowy `(*Message, error)`
We use `Map(ctx context.Context, keys []string, datum Datum) Messages` for our other functions as well. Any specific reason for this change?

Reply:
> let's not return error. what is numaflow going to do if we see an error?

user's might expect error handling, which includes retries etc. it might be better to remove error.

Reply:
It's tricky without having an error returned. 

Assume use do error handling - if he/she logs the error without retry (or after several retry), he/she has to return something, and we need to use the returned value to tell if there's any data returned. Or the user has to keep retrying until the gPRC timeouts, which is not ideal.

Reply:
we are not returning error in other places, e.g., `map`; or are we?

Reply:
> Also @whynowy `(*Message, error)` We use `Map(ctx context.Context, keys []string, datum Datum) Messages` for our other functions as well. Any specific reason for this change?

This is for returned value.

Other functions do not expect user to return errors, and it's user's responsibility to process the message, or we block. This one is different, imagine if there's not a sidecar container for retrieve the data, but do it in the numa container, what we should do when there's an problem? We might retry several times, if it still fails, we log it and wait for next run, and do not update the data store. Now, with this happening in the ud container, we would expect this logic still exits. If the the interface does not return an error, and expect the user to retry forever, then we have to wait for the gPRC timeout error (which will be long).